tools/windows-offset-finder/README.md

```diff
@@ -0,0 +1,217 @@
+# Windows Offset Finder Tools
+- Daniel English
+- John Maccini
+- Mathieu Tarral
+
+# Method 1 - Using Rekall
+
+The [Rekall](https://github.com/google/rekall) framework already integrate an
+address space to work on top of [LibVMI](https://github.com/libvmi/libvmi).
+
+Rekall will find the right GUID, download the PDBs and extract the right offsets for us.
+
+## Requirements
+
+- `LibVMI` C library
+- `libvmi` Python bindings
+- `cabextract`
+
+## Installing Rekall from source
+
+At the time of this writing, there is no release of Rekall available which integrates
+the new address space. Therefore you must install Rekall from source:
+
+    virtualenv -p python3 venv
+    source venv/bin/activate
+    (venv) pip install --upgrade setuptools pip wheel
+    (venv) git clone https://github.com/google/rekall.git
+    (venv) pip install --editable rekall/rekall-lib
+    (venv) pip install --editable rekall/rekall-core
+    (venv) pip install --editable rekall/rekall-agent
+    (venv) pip install --editable rekall
+
+## Usage
+
+The script `rekall_offset_finder.py` takes one argument, the `URL`, indicating how
+Rekall should access the domain.
+Examples:
+
+    (venv) ./rekall_offset_finder.py vmi:///windows_7
+
+You can specify the hypervisor if you want
+
+    (venv) ./rekall_offset_finder.py vmi://kvm/windows_7
+    (venv) ./rekall_offset_finder.py vmi://xen/windows_7
+
+
+Running the script should extract the offset and display a config entry to copy paste.
+Example run:
+
+
+    (venv) ./rekall_offset_finder.py vmi://kvm/win7x64
+    LibVMI Version 0.11.0
+    LibVMI Driver Mode 1
+    --completed driver init.
+    --got id from name (win7x64 --> 10)
+    **set image_type = win7x64
+    --libvirt version 1003001
+    --qmp: virsh -c qemu:///system qemu-monitor-command nitro_win7x64 '{"execute": "pmemaccess", "arguments": {"path": "/tmp/vmi0xcJEx"}}'
+    --kvm: using custom patch for fast memory access
+    **set allocated_ram_size = 59700000, max_physical_address = 0x59700000
+     Trying to fetch http://msdl.microsoft.com/download/symbols/ntkrnlmp.pdb/F8E2A8B5C9B74BF4A6E4A48F180099942/ntkrnlmp.pdb
+
+    nitro_win7x64 {
+        ostype = "Windows";
+        win_pdbase = 0x28;
+        win_pid = 0x180;
+        win_tasks = 0x188;
+        win_pname = 0x2e0;
+        rekall_profile = "/home/wenzel/libvmi/tools/windows-offset-finder/win7x64-profile.json";
```

Comment:
When the config defines `rekall_profile` there is no need to define the other offsets, those are automatically extracted from the profile.

Reply:
I deleted the corresponding output, leaving only the rekall_profile:
~~~
 Trying to fetch http://msdl.microsoft.com/download/symbols/ntkrnlmp.pdb/F8E2A8B5C9B74BF4A6E4A48F180099942/ntkrnlmp.pdb
                                                                                                                                                                    
xenwin7 {
    ostype = "Windows";
    rekall_profile = "/home/tarrma/fsecure/libvmi/tools/windows-offset-finder/xenwin7-profile.json";
}
~~~

But i also left a part of the code commented.
People might want to uncomment it to get the offsets if they don't want to have a rekall json profile file associated to move around.

I find the offsets configuration very handy because i can just keep everything in my `/etc/libvmi.conf`.